tests/run-make/invalid-tmpdir-env-var/rmake.rs

```diff
@@ -0,0 +1,18 @@
+// When the TMP or TMPDIR variable is set to an invalid or non-existing directory,
+// this used to cause an internal compiler error (ICE). After the addition of proper
+// error handling in #28430, this test checks that the expected message is printed.
+// See https://github.com/rust-lang/rust/issues/14698
```

Comment:
Suggestion: please specify *which* env vars influences `std::env::temp_dir`:

- `TMPDIR` on Unix (Linux, macOS)
- `TMP` > `TEMP` > `USERPROFILE` > `C:\Windows` on Windows (`>` indicates fallback precedence if the env var on LHS is not set)